crates/lib3h/src/gateway/gateway_transport_send.rs

```diff
@@ -176,6 +176,21 @@ impl P2pGateway {
         expires_at: std::time::Instant,
         cb: SendCallback,
     ) -> GhostResult<()> {
+        if !self.message_encoding.is_attached() {
+            // we must be in a call chain resulting from an incoming message
+            // we need to wait for the next process() call to continue
+            return self.priv_send_queue_pending(SendMetaData {
+                send_data: SendData::WithFullLowUri(send_data),
+                // we want to invoke this on the very next process call
+                // set our last_attempt back far enough to ensure this
+                last_attempt: std::time::Instant::now()
```

Comment:
Should we be able to pass in None to `priv_send_queue_pending` so that it can do this if we want a "send immediate
"?

Reply:
The option was a little awkward, took a compromise approach and made a helper function to at least increase readability.